src/vulkan-renderer/input/keyboard_mouse_data.cpp

```diff
@@ -1,41 +1,60 @@
 #include "inexor/vulkan-renderer/input/keyboard_mouse_data.hpp"
 
+#include <GLFW/glfw3.h>
+
+#include <shared_mutex>
+
 namespace inexor::vulkan_renderer::input {
 
+KeyboardMouseInputData::KeyboardMouseInputData() {
+    std::unique_lock lock(m_input_mutex);
+    /// We store the pressed keys in a vector of size GLFW_KEY_LAST.
+    m_pressed_keys.resize(GLFW_KEY_LAST);
+    m_pressed_mouse_buttons.resize(GLFW_KEY_LAST);
+}
+
 void KeyboardMouseInputData::press_key(const std::int32_t key) {
+    std::unique_lock lock(m_input_mutex);
```

Comment:
```suggestion
    std::scoped_lock lock(m_input_mutex);
```
Same for the other uses of `std::unique_lock`. They basically do the same thing - but I think it's generally preferred to use `std::scoped_lock` unless you really need the extra features of `std::unique_lock`.

Reply:
Done.